### DISCUSSION TITLE
Software Item Type Improvements

### DISCUSSION THREAD

**Discussion by tmorrell** [inveniosoftware/product-rdm#157] [Ideas]
This issue collects improvements to InvenioRDM for better collecting and preserving research software. It includes porting existing features from Zenodo and improvements suggested by Software Heritage, Caltech Library, and the SciCodes consortium. 

**Deposit Form**

- [x] Add software section to deposit form
- [ ] Make software section responsive (expands when software resource type is selected)
- [ ] Adjust default license when software is selected
- [x] Add Code repository (url) field
- [ ] Add link to GitHub integration and documentation page on software metadata
- [ ] Add subjects vocabulary for operating system
- [ ] Add subjects vocabulary for runtime platform
- [ ] Add subjects vocabulary programming language
- [x] Add subjects vocabulary for development status (from repostatus.org)
- [ ] Add Release notes as a type of additional description
- [ ] Add Software-specific alternative identifier types (e.g. ASCL ID, SWHID)

**Landing Page**

- [x] Add CodeMeta as a downloadable metadata format. 
- [x] Add CITATION.cff as a downloadable metadata format. 
- [x] For BibLateX export, use type @software

**GitHub Integration**
- [x] Port GitHub integration #25 
- [ ] Add CodeMeta as an input format
- [ ] Update GitHub integration metadata mapping
- [ ] Create a new “software PID” API endpoint
- [ ] Update GitHub integration to use "software PID" API endpoint
- [ ] Improve error handling. 
- [ ] Create a GitHub action that prepares a repository for release (by adding in a new DOI)

**Blue sky improvements**
- [ ] Create a InvenioRDM record from a previous GitHub release
- [ ] A button in the GitHub integration panel to generate a pull request to add a CITATION.cff or codemeta file to the source GitHub repository

**Comment by ppanero** [comment]
A first **super dummy and incomplete** version of this has been deployed in https://inveniordm-qa.web.cern.ch/uploads/new using custom fields (see "Software" section at the bottom of the deposit form)

**Comment by moranegg** [comment]
Development is moving forward on https://github.com/inveniosoftware/invenio-swh/
